src/plugins/discover/public/application/view_components/canvas/discover_table_service.tsx

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React, { useEffect } from 'react';
+import { IndexPattern } from '../../../opensearch_dashboards_services';
+import { DiscoverServices } from '../../../build_services';
+import { SavedSearch } from '../../../saved_searches';
+import { useDiscoverTableService } from './utils/use_discover_canvas_service';
+import { DiscoverTableApplication } from './discover_table_app';
+
+export interface DiscoverTableAppProps {
+  services: DiscoverServices;
+  savedSearch: SavedSearch;
+  indexPattern: IndexPattern;
+}
+
+export const DiscoverTableService = ({
```

Comment:
Do we need this? Right now the React tree looks like:

`Canvas` -> `DiscoverTable` -> `DiscoverTableService` -> `DiscoverTableApplication` -> `DataGrid`

Cant we instead simplify this to just be 

`Canvas` -> `DiscoverTable` -> `DataGrid`?

Reply:
This DiscoverTableApplication actually need to rename. It will include DiscoverChart (HitsCount, TimeChartHeader and Histogram) and DiscoverTable.
This discover_table_service is used to have any updates on services, like data & saved search for both DiscoverChart and DiscoverTable. 

We could re-evaluate and simply after canvas parts all implemented.